test/getCommands.spec.js

```diff
@@ -7,136 +7,147 @@ const sinon = require('sinon');
+     */
+    beforeEach(() => {
+      mock(APP_JSON, NO_PLUGINS_JSON);
     });
 
-    mock('rnpm-plugin-test-2', commands.single);
-    mock(path.join(LOCAL_NODE_MODULES, 'rnpm-plugin-test-2'), commands.single);
+    after(() => revert());
 
-    expect(getCommands().length).to.be.equal(1);
-  });
+    it('list of the commands should be a non-empty array', () => {
+      mock(LOCAL_RNPM_PJSON, SAMPLE_RNPM_JSON);
+      mock(SAMPLE_RNPM_PLUGIN, commands.single);
 
-  describe('in local installation', () => {
+      expect(getCommands()).to.be.not.empty;
+      expect(getCommands()).to.be.an('array');
+    });
 
-    var getCommands;
-    var revert;
+    it('should export one command', () => {
+      mock(LOCAL_RNPM_PJSON, SAMPLE_RNPM_JSON);
+      mock(SAMPLE_RNPM_PLUGIN, commands.single);
 
-    before(() => {
-      getCommands = rewire('../src/getCommands');
-      revert = getCommands.__set__('__dirname', path.join(LOCAL_NODE_MODULES, 'rnpm/src'));
+      expect(getCommands().length).to.be.equal(1);
     });
 
-    it('should load rnpm plugins', () => {
-      mock('rnpm-plugin-global', commands.single);
-      mock(appJSON, {});
-      mock(path.join(LOCAL_NODE_MODULES, 'rnpm/package.json'), {
-        dependencies: {
-          'rnpm-plugin-global': '*',
-        },
-      });
+    it('should export multiple commands', () => {
+      mock(LOCAL_RNPM_PJSON, SAMPLE_RNPM_JSON);
+      mock(SAMPLE_RNPM_PLUGIN, commands.multiple);
 
-      expect(getCommands()[0]).to.be.equal(commands.single);
+      expect(getCommands().length).to.be.equal(2);
     });
 
-    it('should load app specific plugins', () => {
-      mock(path.join(LOCAL_NODE_MODULES, 'rnpm-plugin-local-app-plugin'), commands.single);
-      mock(GLOBAL_RNPM_PJSON, {});
-      mock(appJSON, {
+    it('should export unique list of commands by name', () => {
+      mock(LOCAL_RNPM_PJSON, {
         dependencies: {
-          'rnpm-plugin-local-app-plugin': '*',
+          [SAMPLE_RNPM_PLUGIN]: '*',
+          [`${SAMPLE_RNPM_PLUGIN}-2`]: '*',
         },
       });
 
-      expect(getCommands()[0]).to.be.equal(commands.single);
+      mock(SAMPLE_RNPM_PLUGIN, commands.single);
+      mock(`${SAMPLE_RNPM_PLUGIN}-2`, commands.single);
+
+      expect(getCommands().length).to.be.equal(1);
     });
 
-    after(() => revert());
   });
 
-  describe('in global installation', () => {
+  describe('project plugins', () => {
+
     var getCommands;
     var revert;
 
     before(() => {
       getCommands = rewire('../src/getCommands');
-      revert = getCommands.__set__('__dirname', path.join(GLOBAL_NODE_MODULES, 'rnpm/src'));
     });
 
-    it('should load rnpm own plugins', () => {
-      mock('rnpm-plugin-global', commands.single);
-      mock(appJSON, {});
-      mock(GLOBAL_RNPM_PJSON, {
-        dependencies: {
-          'rnpm-plugin-global': '*',
-        },
-      });
-
-      expect(getCommands()[0]).to.be.equal(commands.single);
```

Comment:
We don't have to test for rnpm plugins in this suite since we already tested it in previous one.